tests/testament/htmlgen.nim

```diff
@@ -92,8 +77,12 @@ type
 
 proc allTestResults(): AllTests =
   result.data = newJArray()
-  for file in os.walkFiles("testresults/*.json"):
-    let data = parseFile(file)
+  for file in os.walkFiles("testresults"/"*.json"):
+    var data: JsonNode
+    try: data = parseFile(file)
```

Comment:
No. Let it crash! Don't make this robust for broken test results!

Reply:
@Araq Normally I'd agree, but in this case a malformed JSON testresult file might actually be fairly common. Consider the case where someone starts running all tests, and then presses Ctrl+C. If testament was just in the middle of a testrun, you'd get a half-way written JSON file.

Using this try-catch, you'd still be able to get HTML output for all tests categories that completed before pressing Ctrl+C.

Reply:
When you press Ctrl+C you can delete the broken file before running ``tester html``.

Reply:
Still annoying, with parallel categories you'll have multiple such files!

Reply:
Well we disagree then. But it's not negotiable, please get rid of this hack.